libnmstate/nm/vxlan.py

```diff
@@ -0,0 +1,85 @@
+#
+# Copyright (c) 2019 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from libnmstate.nm import connection
+from libnmstate.nm import nmclient
+from libnmstate.schema import VXLAN
+
+
+def create_setting(iface_state, base_con_profile):
+    vxlan = iface_state.get(VXLAN.CONFIG_SUBTREE)
+    if not vxlan:
+        return None
+
+    vxlan_setting = None
+    if base_con_profile:
+        vxlan_setting = base_con_profile.get_setting_vxlan()
+        if vxlan_setting:
+            vxlan_setting = vxlan_setting.duplicate()
+
+    if not vxlan_setting:
+        vxlan_setting = nmclient.NM.SettingVxlan.new()
+
+    vxlan_setting.props.id = vxlan[VXLAN.ID]
+    vxlan_setting.props.parent = vxlan[VXLAN.BASE_IFACE]
+    vxlan_remote = vxlan.get(VXLAN.REMOTE)
+    if vxlan_remote:
+        vxlan_setting.props.remote = vxlan_remote
+    vxlan_destination_port = vxlan.get(VXLAN.DESTINATION_PORT)
+    if vxlan_destination_port:
+        vxlan_setting.props.destination_port = vxlan_destination_port
+
+    return vxlan_setting
+
+
+def get_info(device):
+    """
+    Provides the current active values for a device
+    """
+    info = {}
+    if device.get_device_type() == nmclient.NM.DeviceType.VXLAN:
+        base_iface = ""
+        if device.props.parent:
+            base_iface = device.props.parent.get_iface()
+        info[VXLAN.CONFIG_SUBTREE] = {
+            VXLAN.ID: device.props.id,
+            VXLAN.BASE_IFACE: base_iface,
+            VXLAN.REMOTE: device.props.group,
+            VXLAN.DESTINATION_PORT: _get_destination_port(device),
+        }
+    return info
+
+
+def _get_destination_port(device):
+    """
+    Retrieve the destination port.
+
+    The destination port is retrieved from the profile settings instead
+    of the device (which represents the kernel state)
+    due to an existing issue [1].
+
+    [1] https://bugzilla.redhat.com/show_bug.cgi?id=1768388
```

Comment:
I noticed the API has `NM.DeviceVxlan.get_dst_port()`. No work?

Reply:
@thom311 fixed it upstream, there where an issue with libnm cache, the fix is at bz.

Reply:
OK. If we know for sure NM 1.22 will fix it, we can add a check on NM version there in case we forgot.
Just a suggestion, not a mast.

Reply:
Well, I don't know when is going to land that, but if you guys know is 1.22 let's check version here with xfail

Thanks!

Reply:
I am not referring to a test case, but:

```
if nm_version >= 1.22:
    NM.DeviceVxlan.get_dst_port()
else:
    use ConnectionProfile

```

Reply:
Yes. It will be fixed in NM 1.22.

Reply:
Done